hogben/tests/test_fisher.py

```diff
@@ -0,0 +1,224 @@
+
+
+def test_fisher_information_values_ref1d(refl1d_model):
+    """
+    Tests that all the calculated Fisher information matrix returns the expected values
+    for a given set of parameters
+    """
+    g = get_fisher_information(refl1d_model, step=0.005)
+    expected_fisher = [
+        [3.58042704e-05, 6.49102395e-05, -4.40696428e-06, -2.83582010e-06],
+        [6.49102395e-05, 1.22021923e-04, -7.66739810e-06, -4.72522420e-06],
+        [-4.40696428e-06, -7.66739810e-06, 6.26586892e-07, 4.56331772e-07],
+        [-2.83582010e-06, -4.72522420e-06, 4.56331772e-07, 3.61390847e-07]]
+    np.testing.assert_allclose(g, expected_fisher, rtol=1e-08, atol=0)
+
+
+@pytest.mark.parametrize('step', (0.01, 0.0075, 0.0025, 0.001, 0.0001))
+def test_fisher_consistent_steps_refnx(step, refnx_model):
+    """
+    Tests whether the Fisher information remains mostly consistent when
+    changing step size using the refnx model
+    """
+    g_reference = get_fisher_information(refnx_model, step=0.005)
+    g_compare = get_fisher_information(refnx_model, step=step)
+    np.testing.assert_allclose(g_reference, g_compare, rtol=1e-02, atol=0)
+
+
+@pytest.mark.parametrize('step', (0.01, 0.0075, 0.0025, 0.001, 0.0001))
+def test_fisher_consistent_steps_refl1d(step, refl1d_model):
+    """
+    Tests whether the Fisher information remains mostly consistent when
+    changing step size using the refl1d model
+    """
+    g_reference = get_fisher_information(refl1d_model, step=0.005)
+    g_compare = get_fisher_information(refl1d_model, step=step)
+    np.testing.assert_allclose(g_reference, g_compare, rtol=1e-02, atol=0)
+
+
+@pytest.mark.parametrize('model_params', (1, 2, 3, 4))
+def test_fisher_shape(model_params):
+    """
+    Tests whether the shape of the Fisher information matrix remains
+     correct when changing the amount of parameters
+    """
+    xi = MockXi([8, 2, 60, 150])[:model_params]
+    bounds = [np.array([6.0, 1.0, 40, 100][:model_params]),
+              np.array([10, 3, 70, 180][:model_params])]
+    expected_shape = (model_params, model_params)
+    g = get_fisher_mock(bounds=bounds, xi=xi)
+    np.testing.assert_array_equal(g.shape, expected_shape)
+
+
+@pytest.mark.parametrize('qs',
+                         (np.arange(0.001, 1.0, 0.25),
+                          np.arange(0.001, 1.0, 0.10),
+                          np.arange(0.001, 1.0, 0.05),
+                          np.arange(0.001, 1.0, 0.01)))
+def test_fisher_diagonal_positive(qs):
+    """Tests whether the diagonal values in the Fisher information matrix
+     are positively valued"""
+    qs = [0.1, 0.2, 0.4, 0.6, 0.8]
+    g = get_fisher_mock(qs=[qs])
+    np.testing.assert_array_less(np.zeros(len(g)), np.diag(g))
+
+
+@pytest.mark.parametrize('xi',
+                         (MockXi([]),
+                          MockXi([8]),
+                          MockXi([8, 2]),
+                          MockXi([8, 2, 60]),
+                          MockXi([8, 2, 60, 150])))
```

Comment:
this can definitely be called in a nicer way, but see my comments about making it a fixture to be called